### DISCUSSION TITLE
Roadmap: Dropwizard Metrics 5.0.x

### DISCUSSION THREAD

**Discussion by hdost** [dropwizard/metrics#2932] [General]
## Remaining Tasks
- [x] #1735
- [x] #1736
- [ ] #2055

**Comment by joschi** [comment]
There are currently no further plans for *Dropwizard Metrics 5.x*.

Once there are, you will be able to see the scheduled work in the [5.0.0 milestone](https://github.com/dropwizard/metrics/milestone/22).

**Comment by hdost** [comment]
@joschi Ok, that's fine, but what I can't find is the reasons there are no further plans. Why did it get paused? What would bring it back? etc.

**Comment by joschi** [comment]
> Ok, that's fine, but what I can't find is the reasons there are no further plans.

There are currently not enough active maintainers and contributors pushing the agenda forward.

> What would bring it back? etc.

Active long-time contributions from interested parties. 😃

CC: @dropwizard/metrics

**Comment by joschi** [comment]
@hdost Sorry that I sounded too harsh.

The issue right now is a lack of people who would be willing to invest substantial time and work into Dropwizard Metrics 5.x.

Past updates with backward-incompatible changes (e. g. Dropwizard Metrics 3.x -> 4.x) have shown that only a small number of people are willing to invest in the update once it's been released.

This leads to the situation that there are a lot of consumers still using an outdated and unsupported version of Dropwizard Metrics: https://twitter.com/dropwizardio/status/1259130162794373122

![metrics-core usage in May 2020](https://user-images.githubusercontent.com/43951/98565978-8838a680-22ae-11eb-8b34-8401dc537751.png)

With this in mind, I'm wondering if investing any time into yet another major version with breaking changes (which would have to be picked up by consumers) currently makes sense.

**Comment by hdost** [comment]
I appreciate the feedback :) we're currently looking at our efforts internally and we're hoping to implement tagging/labels for our metrics, but that's only in the 5.x branch for now.

**Comment by arteam** [comment]
Hey @hdost and @joschi! Artem is here, I originally started the work on Metrics 5 a couple of years ago, but have been working on it only sporadically. I've recently merged the latest changes from the `release/4.1` and `release/4.2` branches to `5.0-development`. The test suite is green now and Metrics 5.0 is usable to integrate directly or via the Metrics4 adapter. I can cut a new release candidate in Maven Central and see what are outstanding issues for Metrics 5. The most interesting use case would be to see how it integrates with Micrometer which already supports tags. Luckily, I'm going to have to have some free time in November - December because of the lockdown and I think it would be a good opportunity to finally release Metrics 5 in the wild!

**Comment by Sharpie** [comment]
Running sums are another feature I'd love to see get an official release: #1254

**Comment by hdost** [comment]
@arteam  is there some sort of investigation or work I could help with?

**Comment by hdost** [comment]
@arteam :wave: I've done a little bit of testing with the micrometer library, but I could also just be missing things it seems to be working alright.

**Comment by sergey-morenets** [comment]
Hi

Do you have any ideas/plans to support Jakarta EE 9 in 5.x branch ?

**Comment by sergey-morenets** [comment]
I noticed that metrics-servlets module in 5.x branch , for example, still uses Servlet API 4:

https://github.com/dropwizard/metrics/blob/5.0-development/metrics-servlets/pom.xml

**Comment by joschi** [comment]
> I noticed that metrics-servlets module in 5.x branch , for example, still uses Servlet API 4:

@sergey-morenets Would you be able to provide a forward-port of https://github.com/dropwizard/metrics/pull/1730 into the `release/5.0.x` branch?

**Comment by joschi** [comment]
@sergey-morenets Or better, port the missing functionality from `release/4.2.x` to `release/5.0.x`?

**Comment by sergey-morenets** [comment]
> 
> 
> @sergey-morenets Or better, port the missing functionality from `release/4.2.x` to `release/5.0.x`?

Thank you for suggestion. But I'm not sure if I have even permissions to work on this project.

**Comment by joschi** [comment]
> Thank you for suggestion. But I'm not sure if I have even permissions to work on this project.

@sergey-morenets You can fork the project and create a pull request. 😃

https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork

**Comment by joschi** [comment]
@arteam Could you please reconcile the `5.0-development` and `release/5.0.x` branches (#1735) and port the changes from `release/4.2.x` to `release/5.0.x` (#1736)?

**Comment by mbrannstrom** [comment]
What is the status of the 5.0.x version? 
Will it ever be released? 

Tagged/Labeled metrics is quite a big thing... and metrics 4.x does not have it.

**Comment by shahbaaz31here** [comment]
I am also looking for an answer for @mbrannstrom  question related to tags feature. Any good news folks ?

**Comment by joschi** [comment]
@shahbaaz31here It's simple: If somebody steps up as a permanent maintainer for the Dropwizard Metrics 5.x branch, we can release it.

**Comment by shahbaaz31here** [comment]
Thanks @joschi . Does anyone know if there is any custom library available for adding tags ? We can use that until 5.x version comes.

**Comment by twcurrie** [comment]
Is there no status change here?  Is Dropwizard Metrics 5.x release still blocked on a having maintainer?

**Reply by joschi** [reply]
> Is Dropwizard Metrics 5.x release still blocked on a having maintainer?

Yes, or in other words the person (or people) maintaining that version and performing the actual release. 😬

**Comment by chaitanyachavali** [comment]
I have been a dropwizard user for almost 5 years now, and I am personally very interested in contributing. Would be very glad to maintain and handle the future releases of [dropwizard/metrics](https://github.com/dropwizard/metrics/).

@joschi Please help me where to start, if you have any documentation around the release process and so on, do share. You might have to bear with me for sometime to pick things up, but I promise I am a quick learner and will get going soon.  Also, I was wondering if this would be more of a release engineering or would it also lead to technical contributions to this or other dropwizard projects.

**Comment by praveen97uma** [comment]
I am interested in maintaining and contributing to metrics and other dropwizard projects as well. Have been an active user of dropwizard for 6+ years now and would love to take this forward. @joschi

**Comment by joschi** [comment]
@chaitanyachavali @praveen97uma Thanks for showing interest in contributing to this project!

The last remaining thing to do for Dropwizard Metrics 5.0.x is described in https://github.com/dropwizard/metrics/issues/2167.

Would you like to give it a try?

**Reply by chaitanyachavali** [reply]
@joschi Sure. 
Completely occupied this week. Will try to go into details of this issue over this weekend or next week.
